### ISSUE TITLE
Binary Accuracy is wrong for other activation function except ReLU !

### ISSUE THREAD

**Issue by saranpan** [wallik2/dipple#2]
Since the problem Grad_check_nn was closed, we are still facing on the following problem
 
we try comparing tanh and ReLU

Training the following hyperparameter setting :
```
1.           {"L" : 5,
              "layer_dims" : [2,8,6,4,2,1],  #
              "lr" : 1e-2,
              "forward_activation_function" : 'relu',
              "last_forward_activation_function" : 'sigmoid'}
```
2.
```
 model = Binary_Deep_L_Layer(hyperparam)
    model.compiles(initialization='he',regularization='L2',lambd= lambd )
    model.fit(X,Y,Epochs=48000,batch_size=32,grad_check=True)
```

On Crescent dataset

Tanh: 
<img width="245" alt="tanh is wrong" src="https://user-images.githubusercontent.com/69796499/180731914-86f02edc-b84c-4a23-9f05-c30fc3bbf5d4.png">
Forgot to capture the loss function at the last epoch, it is about 0.07 

ReLU:
<img width="392" alt="relu is right" src="https://user-images.githubusercontent.com/69796499/180732808-b7852d63-6dd6-42c1-8ada-72e35b7ac028.png">

The accuracy of Tanh was wrong, look at the result of decision boundary, TanH accuracy is worse than ReLU which does not make sense. Something must be wrong

**Comment by saranpan** [comment]
Fixed !! 

The problem was that the function predict_dec

```
def predict_dec(param, X):
    """
    Used for plotting decision boundary.
    
    Arguments:
    model -- expecting model to contain parameter and its activation function 
             where python dictionary containing your param 
    X -- input data of size (m, K)
    
    Returns
    predictions -- vector of predictions of our model (red: 0 / blue: 1)
    """
    
    # Predict using forward propagation and a classification threshold of 0.5

    
    a3, cache = L_model_forward(X, param )
    predictions = (a3 >0.5 ) #Default as 0.5 
    
    return predictions

```

Since it depended on `L_model_forward ` which required the forward_activation_function and last_activation_function as its argument, and we did not specify them, so  `L_model_forward ` assumes that its activation function, last a. is ReLU, sigmoid respectively

So,    instead of 
` def predict_dec(param, X):`

we fixed to 
` def predict_dec(model, X):`
in which model is Dipple model object which contains parameter, activation_function, last_activation_function attributes

So the last fixed function is....
```
def predict_dec(model , X):
    """
    Used for plotting decision boundary.
    
    Arguments:
    model -- expecting model to contain parameter and its activation function 
             where python dictionary containing your param 
    X -- input data of size (m, K)
    
    Returns
    predictions -- vector of predictions of our model (red: 0 / blue: 1)
    """
    
    # Predict using forward propagation and a classification threshold of 0.5
    param = model.param
    forward_activation_function = model.forward_activation_function
    last_forward_activation_function = model.last_forward_activation_function
    
    a3, cache = L_model_forward(X, param , forward_activation_function, last_forward_activation_function)
    predictions = (a3 >0.5 ) #Default as 0.5 
    
    return predictions
```

we also remove the default activation and last activation function of L_model_forward and backward to make sure it does not cause bugs like this anymore

**Comment by saranpan** [comment]
See the results 

LeakyReLU
<img width="279" alt="LeakyReLU" src="https://user-images.githubusercontent.com/69796499/180984111-e0e31652-82a2-414f-911c-fe0d6b1612a6.png">

TanH
<img width="303" alt="TanH" src="https://user-images.githubusercontent.com/69796499/180984159-d597b99f-f26c-4476-b551-c9db781664c7.png">
